src/NHibernate/Loader/OuterJoinableAssociation.cs

```diff
@@ -34,7 +34,12 @@ public sealed class OuterJoinableAssociation
 			rhsColumns = JoinHelper.GetRHSColumnNames(joinableType, factory);
 			on = new SqlString(joinableType.GetOnCondition(rhsAlias, factory, enabledFilters));
 			if (SqlStringHelper.IsNotEmpty(withClause))
-				on = on.Append(" and ( ").Append(withClause).Append(" )");
+			{
+				on = on.Count == 0
+					? on.Append(withClause)
+					: on.Append(" and ( ").Append(withClause).Append(" )");
```

Comment:
For me, that is an undue change. It looks to me the way it is generated, `on` is either empty or starts with `" and "`. Testing on `Count` here changes behavior. Better not do that and keep previous code.